scripts/create-azure-resources.sh

```diff
@@ -0,0 +1,117 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+##--------------------------------------------------------------------
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+: "${AKS_NAME:?AKS_NAME environment variable empty or not defined.}"
+: "${ACR_NAME:?ACR_NAME environment variable empty or not defined.}"
+
+register_feature() {
+  az extension add --name aks-preview
+  az feature register --namespace "Microsoft.ContainerService" --name "EnableWorkloadIdentityPreview"
+  az provider register --namespace Microsoft.ContainerService
+}
+
+create_user_managed_identity() {
+  SUBSCRIPTION_ID="$(az account show --query id --output tsv)"
+
+  az identity create \
+    --name "${USER_ASSIGNED_IDENTITY_NAME}" \
+    --resource-group "${GROUP_NAME}" \
+    --location "${LOCATION}" \
+    --subscription "${SUBSCRIPTION_ID}"
+
+  USER_ASSIGNED_IDENTITY_OBJECT_ID="$(az identity show --name "${USER_ASSIGNED_IDENTITY_NAME}" --resource-group "${GROUP_NAME}" --query 'principalId' -otsv)"
+}
+
+create_acr() {
+  az acr create --name "${ACR_NAME}" \
+    --resource-group "${GROUP_NAME}" \
+    --sku Standard >/dev/null
+  az acr login -n ${ACR_NAME}
+  echo "ACR '${ACR_NAME}' is created"
+
+  # Enable acrpull role to the user-managed identity.
+  az role assignment create \
+    --assignee-object-id ${USER_ASSIGNED_IDENTITY_OBJECT_ID} \
+    --role acrpull \
+    --scope subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${GROUP_NAME}/providers/Microsoft.ContainerRegistry/registries/${ACR_NAME}
+}
+
+create_aks() {
+  az aks create \
+    --resource-group "${GROUP_NAME}" \
+    --name "${AKS_NAME}" \
+    --node-vm-size Standard_DS3_v2 \
+    --kubernetes-version "${KUBERNETES_VERSION}" \
+    --node-count 1 \
+    --generate-ssh-keys \
+    --enable-workload-identity \
+    --attach-acr ${ACR_NAME} \
+    --enable-oidc-issuer >/dev/null
+  echo "AKS '${AKS_NAME}' is created"
+
+  az aks get-credentials --resource-group ${GROUP_NAME} --name ${AKS_NAME}
+  echo "Connected to AKS cluster"
+
+  # Establish federated identity credential between the managed identity, the
+  # service account issuer and the subject.
+  local AKS_OIDC_ISSUER="$(az aks show -n ${AKS_NAME} -g ${GROUP_NAME} --query "oidcIssuerProfile.issuerUrl" -otsv)"
+  az identity federated-credential create \
+    --name ratify-federated-credential \
+    --identity-name "${USER_ASSIGNED_IDENTITY_NAME}" \
+    --resource-group "${GROUP_NAME}" \
+    --issuer "${AKS_OIDC_ISSUER}" \
+    --subject system:serviceaccount:"${RATIFY_NAMESPACE}":"ratify-admin"
+
+  # It takes a while for the federated identity credentials to be propagated
+  # after being initially added.
+  sleep 1m
+}
+
+create_akv() {
+  # az keyvault create \
```

Comment:
I assume this is temporarily commented out?

Reply:
After reviewing further, I see that your plan is to not create keyvault for each for now and then switch to pushing signed images in a follow up PR which will switch to new KV everytime?

Reply:
Correct, I'll delete commented code once I've done my tests.